src/components/DepositWidget/useRowActions.tsx

```diff
@@ -16,13 +17,71 @@ interface Params {
 
 interface Result {
   enableToken: (tokenAddress: string) => Promise<void>
-  deposit: (amount: BN, tokenAddress: string) => Promise<void>
-  requestWithdraw: (amount: BN, tokenAddress: string) => Promise<void>
-  claim: (tokenAddress: string) => Promise<void>
+  depositToken: (amount: BN, tokenAddress: string) => Promise<void>
+  requestWithdrawToken: (amount: BN, tokenAddress: string) => Promise<void>
+  claimToken: (tokenAddress: string) => Promise<void>
+  // State
+  enabling: TokenLocalState['enabling']
+  highlighted: TokenLocalState['highlighted']
+  claiming: TokenLocalState['claiming']
+}
+
+interface TokenLocalState {
+  enabling: Map<string, string>
+  highlighted: Map<string, string>
+  claiming: Map<string, string>
+}
+
+const SET_ENABLING = 'enabling'
+const SET_CLAIMING = 'claiming'
+const SET_HIGHLIGHTED = 'highlighted'
+const SET_HIGHLIGHTED_AND_CLAIMING = 'highlighted_and_claiming'
+
+type Actions =
```

Comment:
This just asks for an enum
```ts
const enum ActionTypes  {
    SET_ENABLING = 'enabling',
    SET_CLAIMING = 'claiming',
    SET_HIGHLIGHTED = 'highlighted',
    SET_HIGHLIGHTED_AND_CLAIMING = 'highlighted_and_claiming',
}

interface Action {
    type: ActionTypes
    payload: string
}
```

though it's very much individual preference

Reply:
me likey